src/main/groovy/com/bmuschko/gradle/docker/DockerExtension.groovy

```diff
@@ -97,6 +97,8 @@ class DockerExtension {
             boolean isWindows = isWindows()
             if (!isWindows && new File('/var/run/docker.sock').exists()) {
                 dockerUrl = 'unix:///var/run/docker.sock'
+            } else if (!isWindows && new File("${System.getProperty("user.home")}/.docker/run/docker.sock").exists()) {
```

Comment:
I'd have a nested if conditional here. That'd make it a bit more readable I think.

Something like:

```
if (!isWindows) {
   if(new File('/var/run/docker.sock').exists()) {
      // ...
   }
   else if(new File("${System.getProperty("user.home")}/.docker/run/docker.sock").exists()) {
       // ...
   }
}
```

Long term, I think we should break out this logic into a different class to make it testable. I'd really want some basic unit tests for this logic. Do you think you could do that? If not, I can work on it later.

Reply:
I can move this out and cover it with some tests.